ui/components/component-library/icon/icon.scss

```diff
@@ -39,4 +39,8 @@
   &--size-xl {
     --size: 32px;
   }
+
+  &--size-auto {
+    --size: 1em;
+  }
```

Comment:
Will create a ticket for this because we will need to add the docs and story https://github.com/MetaMask/metamask-extension/issues/16006